src/parcels/_core/utils/sgrid.py

```diff
@@ -210,6 +230,9 @@ def to_attrs(self) -> dict[str, str | int]:
             volume_dimensions=dump_mappings(self.volume_dimensions),
         )
 
+    def rename_dims(self, dims_dict: dict[str, str]) -> Self:
```

Comment:
Just for my own learning, why does this method return a `Self` with a capital 'S'?